automount-actions/scripts/bin/auto-install.sh

```diff
@@ -147,21 +148,19 @@ validate_snap_file() {
     fi
     
     # Check if it's a SquashFS file (snap files are SquashFS archives)
-    # SquashFS magic number is 'hsqs' (0x73717368) at offset 0
-    if command -v hexdump >/dev/null 2>&1; then
-        _magic=$(hexdump -C "$_snap_file" 2>/dev/null | head -1 | cut -d' ' -f2-5 | tr -d ' ')
-        if [ -n "$_magic" ] && [ "$_magic" != "68737173" ]; then
-            log "WARN: Snap file does not appear to be SquashFS format: $_snap_file"
-            # Don't fail here as snapd will do final validation
-        fi
+    # SquashFS magic number is 'hsqs' at offset 0
+    # Use dd to read first 4 bytes, redirect stderr to avoid noise
+    _magic=$(dd if="$_snap_file" bs=4 count=1 2>/dev/null)
+    if [ -n "$_magic" ] && [ "$_magic" != "hsqs" ]; then
```

Comment:
It feels more correct to check:
```suggestion
[ -z "$_magic" ] || [ "$_magic" != hsqs ] && { ... }
```

Reply:
The following pattern is undesirable, because if `is_condition_met` is false, and we've `set -e` (like we're supposed to) then the script will immediately exit non-zero.
```sh
is_condition_met && {
  : # Do some stuff
}
```

You'll want to invert the logic (which can also be simplified):
```sh
[ "${_magic:-}" = hsqs ] || {
  : # Do some stuff
}
```

Reply:
I always manage to muck about... >.< Doesn't shellcheck usually emit a warning for this? It didn't when I wrote it /shrug

I can't imagine a scenario where we need to check `-z|-n` anyways when we're already checking for a string; we don't care so much about what it is if it isn't hsqs.

Reply:
hmm.. not sure, but it's possible that this is legitimately desirable behavior in some cases, which means shellcheck wouldn't complain.

Agreed on `-z`/`-n`.